src/software/pipeline/main_panoramaWarping.cpp

```diff
@@ -975,28 +979,21 @@ bool computeOptimalPanoramaSize(std::pair<int, int> & optimalSize, const sfmData
 
   
   if (scales.size() > 1) {
-    double median_scale;
-    std::nth_element(scales.begin(), scales.begin() + scales.size() / 2, scales.end());
-    median_scale = scales[scales.size() / 2];
 
-    double multiplier = pow(2.0, int(floor(log2(median_scale))));
-    
+    std::sort(scales.begin(), scales.end());
+    int selected_index = int(floor(float(scales.size() - 1) * ratioUpscale));
+    double selected_scale = scales[selected_index];
+
+    ALICEVISION_LOG_INFO("Computed panorama size: "  << optimalSize.first * selected_scale << "x" << optimalSize.second * selected_scale);
```

Comment:
We may need to cast to integers to avoid floating point values in log.

Reply:
will do